src/decode.rs

```diff
@@ -163,3 +265,34 @@ mod tests {
         assert_eq!(decode("a3gV").into(&mut output), Err(DecodeError::BufferTooSmall));
     }
 }
+
+#[cfg(test)]
+#[cfg(feature = "check")]
+mod test_check{
+    use decode;
+    use decode::DecodeError;
+
+    #[test]
+    fn test_check(){
+        for &(val, s) in super::super::CHECK_TEST_CASES.iter() {
+            assert_eq!(val.to_vec(), decode(s).with_check(None).into_vec().unwrap());
+        }
+
+        for &(val, s) in super::super::CHECK_TEST_CASES[1..].iter() {
+            assert_eq!(val.to_vec(), decode(s).with_check(Some(val[0])).into_vec().unwrap());
+        }
+    }
+
+    #[test]
+    fn test_check_ver_failed() {
+        let d = decode(super::super::CHECK_TEST_CASES[6].1)
+            .with_check(Some(0x01))
+            .into_vec();
+
+        assert!(d.is_err());
+        if let DecodeError::InvalidVersion {ver: _, expected_ver: _} = d.unwrap_err() {}
```

Comment:
Even if it's just for this one test, I think it's worth it to bring in the [`assert_matches`](https://crates.io/crates/assert_matches) crate for this.

Reply:
Done